test/powershell/Language/Interop/DotNet/DotNetInterop.Tests.ps1

```diff
@@ -180,7 +225,95 @@ namespace DotNetInterop
         }
     }
 
-    It "Set access of an indexer that accepts ByRef-like type should fail gracefully" {
-        { $testObj[1] = 1 } | Should -Throw -ErrorId "InvalidCastToByRefLikeType"
+    It "Set access of an indexer that accepts ByRef-like type value should fail gracefully" {
+        { $testObj[1] = 1 } | Should -Throw -ErrorId "CannotIndexWithByRefLikeReturnType"
+    }
+
+    Context "Passing value that is implicitly/explicitly castable to ByRef-like parameter in method invocation" {
+        
+        BeforeAll {
+            $typeXml = @'
+<Types>
+    <Type>
+        <Name>DotNetInterop.Test2</Name>
+        <Members>
+            <CodeMethod>
+                <Name>RunTest</Name>
+                <CodeReference>
+                    <TypeName>DotNetInterop.CodeMethods</TypeName>
+                    <MethodName>RunMethod</MethodName>
+                </CodeReference>
+            </CodeMethod>
+            <CodeProperty>
+                <Name>TestName</Name>
+                <GetCodeReference>
+                    <TypeName>DotNetInterop.CodeMethods</TypeName>
+                    <MethodName>GetProperty</MethodName>
+                </GetCodeReference>
+                <SetCodeReference>
+                    <TypeName>DotNetInterop.CodeMethods</TypeName>
+                    <MethodName>SetProperty</MethodName>
+                </SetCodeReference>
+            </CodeProperty>
+        </Members>
+    </Type>
+</Types>
+'@
+            $typeXmlFile = Join-Path -Path $TestDrive -ChildPath "TestType.ps1xml"
+            Set-Content -Path $typeXmlFile -Value $typeXml -Encoding Ascii
+            $ps = [powershell]::Create()
+            $ps.AddCommand("Update-TypeData").AddParameter("AppendPath", $typeXmlFile).Invoke()
+            $ps.Commands.Clear()
+            $ps.AddScript('$test = [DotNetInterop.Test2]::new()').Invoke()
+            $ps.Commands.Clear()
+        }
+
+        AfterAll {
+            $ps.Dispose()
+        }
+
+        It "Support method calls with ByRef-like parameter as long as the argument can be casted to the ByRef-like type" {
+            $testObj.PrintMySpan("abc", "def") | Should -BeExactly "abc3"
+            $testObj.PrintMySpan("abc", "Hello".ToCharArray()) | Should -BeExactly "abc5"
+            { $testObj.PrintMySpan("abc", 12) } | Should -Throw -ErrorId "MethodArgumentConversionInvalidCastArgument"
+
+            $path = [System.IO.Path]::GetTempPath()
+            [System.IO.Path]::IsPathRooted($path.ToCharArray()) | Should -Be $true
+        }
+
+        It "Support constructor calls with ByRef-like parameter as long as the argument can be casted to the ByRef-like type" {
+            $result = [DotNetInterop.Test2]::new("abc")
+            $result.Name | Should -BeExactly "Number of chars: 3"
+
+            { [DotNetInterop.Test2]::new(12) } | Should -Throw -ErrorId "MethodCountCouldNotFindBest"
```

Comment:
The same.

Reply:
`new(12)` is to use an integer to test the "not castable" scenario. What is your suggestion to change?